frontend/src/app/api/attachment-download/[opportunityId]/route.ts

```diff
@@ -0,0 +1,81 @@
+import { randomUUID } from "crypto";
+import * as zip from "@zip.js/zip.js";
+import { ApiRequestError, readError } from "src/errors";
+import { getOpportunityDetails } from "src/services/fetch/fetchers/opportunityFetcher";
+
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: Promise<{ opportunityId: string }> },
+): Promise<Response> {
+  const { opportunityId } = await params;
+  try {
+    const response = await getOpportunityDetails(opportunityId);
+
+    if (!response || response.status_code !== 200) {
+      throw new ApiRequestError(
+        `Error fetching Opportunity: ${response.message}`,
+        "APIRequestError",
+        response.status_code,
+      );
+    }
+
+    if (response.data.attachments?.length === 0) {
```

Comment:
[nit] I think the following will clean up the logic a little bit further down (no need for the optional chain on line 40)
```suggestion
    if (!response.data.attachments || !response.data.attachments.length) {
```